cms/djangoapps/contentstore/rest_api/v1/views.py

```diff
@@ -0,0 +1,112 @@
+from rest_framework.response import Response
+
+from common.lib.xmodule.xmodule.course_module import get_available_providers
+from contentstore.views.course import get_course_and_check_access
+from models.settings.course_metadata import CourseMetadata
+from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
+from openedx.core.lib.api.view_utils import view_auth_classes
+from xmodule.modulestore.django import modulestore
+
+
+from .serializers import ProctoredExamSettingsMainSerializer
+
+
+@api_view(['GET'])
+@authentication_classes([
+    JwtAuthentication,
+    BearerAuthenticationAllowInactiveUser,
+    SessionAuthenticationAllowInactiveUser,
+])
+@permission_classes([permissions.JWT_RESTRICTED_APPLICATION_OR_USER_ACCESS])
+def proctored_exam_settings(request, course_id):
+    """
+    A view for retrieving information about proctored exam settings for a course.
+
+    Path: ``/api/contentstore/v1/proctored_exam_settings/{course_id}``
+
+    Accepts: [GET]
+
+    ------------------------------------------------------------------------------------
+    GET
+    ------------------------------------------------------------------------------------
+
+    **Returns**
+
+        * 200: OK - Contains a set of course proctored exam settings.
+        * 401: The requesting user is not authenticated.
+        * 403: The requesting user lacks access to the course.
+        * 404: The requested course does not exist.
+
+    **Response**
+
+        In the case of a 200 response code, the response will proctored exam settings data
+        as well as other metadata about the course or the requesting user that are necessary
+        for rendering the settings page.
+
+    **Example**
+
+        {
+            "proctored_exam_settings": {
+                "enable_proctored_exams": true,
+                "allow_proctoring_opt_out": true,
+                "proctoring_provider": "mockprock",
+                "proctoring_escalation_email": null,
+                "create_zendesk_tickets": true
+            },
+            "available_proctoring_providers": [
+                "mockprock",
+                "proctortrack"
+            ],
+            "course_start_date": "2013-02-05T05:00:00Z",
+            "is_staff": true
+        }
+    """
+    course_key = CourseKey.from_string(course_id)
+    with modulestore().bulk_operations(course_key):
+        if request.method == 'GET':
+            course_module = get_course_and_check_access(course_key, request.user)
+
+            if not course_module:
+                return Response(
+                    'Course with course_id {} does not exist.'.format(course_id),
+                    status=status.HTTP_404_NOT_FOUND
+                )
+
+            course_metadata = CourseMetadata().fetch_all(course_module)
+            data = {}
+
+            # specify only the advanced settings we want to return
+            proctored_exam_settings_advanced_settings_keys = [
```

Comment:
While I don't want to block your effort, do realize that the storage of these course-administration settings within Studio is counter to the directional separation we want to have between Studio and LMS, per [this ADR](https://github.com/edx/edx-platform/blob/master/docs/decisions/0005-studio-lms-subdomain-boundaries.rst).

Reply:
Thank you for surfacing this! We're not intending to add any more data to Studio; we are just exposing the ability to view and modify it via a better user interface. I realize that this may not belong in the content authoring subdomain in the future, but the movement of the new user interface to another subdomain (i.e. course run administration) will be easier with this work. Just to clarify, do you have any particular concerns that we should be aware of, or is this to let us know that the storage of course administration settings in Studio is not something we want to maintain for the longterm? 

I read through the ADR, and I didn't see that separation mentioned explicitly. Particularly, when reading the following paragraph,
 
> Note that it is likely that we'll further separate content authoring (e.g. content libraries) from course run administration (e.g. grading policy). It's possible that both of these will evolve under the umbrella of what users see as "Studio". Even if that happens, there will still be an architectural split between the Content Authoring and Learning subdomains within that new Studio.

it's not clear to me where content administration belongs in the current landscape, if not in Studio. I understand that, in the future, from the perspective of a user, course run administration (which is where I am assuming proctoring settings will belong) may very well fall into the Studio "umbrella", but that experience may not be reflective of the eventual underlying architecture, in which there will be a split between content authoring and course run administration. But in the absence of that architectural distinction now, I don't think that this work really pushes it in the diametrical extreme. Am I missing something?

Reply:
@MichaelRoytman Thanks for following up to help me clarify. You're right that the ADR is not as explicit about this distinction between content authoring and course run administration (cc: @ormsbee) - though the paragraph you quote mentions a possible future.

Today, the only dedicated location we have for course run administration is the instructor dashboard. This too will eventually be rehauled.

The reason for my pointing this out was to remind the team that we do plan to eventually have a clearer split between LMS and Studio - and in that future, _"LMS data should not flow backward into Studio"_.  If we are thinking of Proctoring as a course-run administration tool, then we'll want to configure it within LMS itself.

But as you point out, the current work is making use of data that already exists today in Studio. So the team can consider when and how to transition to our desired split - if not during this current effort.

Reply:
By the way, storing these settings in SQL, rather than modulestore, could be an incremental step forward to helping us migrate in the future.  Once again, not suggesting that the team take on this work right now - but something for the team to consider amongst other priorities.